### DISCUSSION TITLE
Problem with two-way binding - v-model?

### DISCUSSION THREAD

**Discussion by Schetnan** [nuxt/nuxt#18038] [Questions]
I am wondering if there is an issue with two way binding with nuxt3's current state?  I have the following code.  The default initialized data shows up in the input field and the control.  However, if I change the data in the field it does not update the field.  Am I missing something, or are there currently issues with two way data binding?

```
<template>
   <div>
     <input type="text" v-model="testme"/>
    {{testme}}
  </div>
</template>

<script>
  export default {
      setup() {
          const testme = ref('test')
          return { testme }
      },
  }
</script>
```

**Comment by Schetnan** [comment]
I figured out my issue.  There was a problem in my setup function in my app.vue ( not the page with the above code ).  I noticed a warning ( not an error ) in the browser console that led me to the problem.  After fixing the setup issue in app.vue, two-way data-binding started working.

**Reply by Schetnan** [reply]
I spoke too soon.  I was able to get this working if running http://localhost:3000.  However, we tie our applications together in development AND production through a proxy which also injects a web component library, and header/footer.  When I run through the proxy, it finds our app let say at /my/app since we have told the proxy listening at https://localhost:8843/ to proxy to https://localhost:3000 when it finds the /my/app path.  The problem is that the Nuxt3 development environment thinks  it is still going through the root host and trying to load _nuxt files at /_nuxt and 404s are getting returned since there is no mapping to them.  It appears that some important files are not being loaded which affects the data-binding.  We had a similar issue with Nuxt2, but the only drawback we have experienced is the need to refresh the page after a code change.  With Nuxt3 it seems more complicated.  This could be a showstopper for us.

I am wondering.  Is there a way to tell the Nuxt3 dev environment to run everything with a particular context root such as /my/app where all files - pages and system _nuxt files get loaded through the same root?

Or, is it possible to tell the nuxt3 development environment to not be quite so fancy and just deliver everything the app needs all at once?

**Reply by mikespinks0401** [reply]
I just ran into this issue, did this end up getting resolved?

**Reply by ShaggyTech** [reply]
Maybe you want to set the router base url or public base path?  I believe in Nuxt 3 you do this through Vite config, maybe this is will provide further info: [ vite docs public base path](https://vitejs.dev/guide/build.html#public-base-path)

If you need to know how to set vite options in nuxt 3 - [nuxt 3 docs](https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#vite)

In Nuxt 2 you could do this by setting `router: {base: “/some-path”}` in the nuxt.config.  I used it this way for Github-Pages deployments which needed the base set to the github package(repo) name.
